### PR TITLE
Feature/wallet connect in assets

### PR DESCRIPTION
**Pull request by antonijzelinskij**

### CODE REVIEW

File: app/src/main/java/io/novafoundation/nova/app/root/presentation/RootViewModel.kt [hunk 1]
```diff
@@ -55,6 +56,10 @@ class RootViewModel(
         syncCurrencies()
 
         updatePhishingAddresses()
+
+        walletConnectService.setOnPairErrorCallback {
+            showError(it)
+        }
```
Comment (valentunn) [inline app/src/main/java/io/novafoundation/nova/app/root/presentation/RootViewModel.kt:62]: I would rather suggest to expose LiveData<Event<Error>> in `WalletConnectService` and to subscribe to it in `rootViewModel`

File: common/src/main/java/io/novafoundation/nova/common/base/BaseActivity.kt [hunk 2]
```diff
@@ -44,15 +41,13 @@ abstract class BaseActivity<T : BaseViewModel> :
...
         subscribe(viewModel)
-    }
-
-    abstract fun inject()
 
-    abstract fun layoutResource(): Int
+        viewModel.errorLiveData.observeEvent(::showError)
 
-    abstract fun initViews()
+        viewModel.messageLiveData.observeEvent(::showMessage)
```
Comment (valentunn) [inline common/src/main/java/io/novafoundation/nova/common/base/BaseActivity.kt:47]: Why do we now subscribe in `attachBaseContext`?

File: feature-account-api/src/main/java/io/novafoundation/nova/feature_account_api/view/WalletConnectView.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import androidx.core.view.isVisible
+import io.novafoundation.nova.common.view.shape.addRipple
+import io.novafoundation.nova.common.view.shape.getRoundedCornerDrawable
+import io.novafoundation.nova.common.view.shape.getRoundedCornerDrawableFromColors
+import io.novafoundation.nova.feature_account_api.R
+import kotlinx.android.synthetic.main.view_wallet_connect.view.viewWalletConnectConnectedCount
+import kotlinx.android.synthetic.main.view_wallet_connect.view.viewWalletConnectConnectionsIcon
+import kotlinx.android.synthetic.main.view_wallet_connect.view.viewWalletConnectIconContainer
+
+class WalletConnectView @JvmOverloads constructor(
```
Comment (valentunn) [inline feature-account-api/src/main/java/io/novafoundation/nova/feature_account_api/view/WalletConnectView.kt:17]: I suggest to name it more specifically. Maybe `WalletConnectCounterView` view?

File: feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/list/BalanceListViewModel.kt [hunk 4]
```diff
@@ -194,6 +207,17 @@ class BalanceListViewModel(
         router.openAssetSearch()
     }
 
+    fun walletConnectClicked() {
+        launch {
+            if (walletConnectAccountSessions.first() > 0) {
```
Comment (valentunn) [inline feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/list/BalanceListViewModel.kt:212]: `walletConnectAccountSessions.first().isNotEmpty()`

File: feature-wallet-connect-impl/src/main/java/io/novafoundation/nova/feature_wallet_connect_impl/domain/session/RealWalletConnectSessionInteractor.kt [hunk 5]
```diff
@@ -136,6 +136,15 @@ class RealWalletConnectSessionInteractor(
         return walletConnectSessionRepository.getSessionAccount(sessionTopic)
     }
 
+    override fun activeSessionsFlow(metaId: Long): Flow<List<WalletConnectSession>> {
+        return walletConnectSessionRepository.sessionAccountsByMetaIdFlow(metaId).map { sessionAccounts ->
+            val activeSessions = Web3Wallet.getListOfActiveSessions()
+            val metaAccount = accountRepository.getMetaAccount(metaId)
```
Comment (valentunn) [inline feature-wallet-connect-impl/src/main/java/io/novafoundation/nova/feature_wallet_connect_impl/domain/session/RealWalletConnectSessionInteractor.kt:142]: Its better to subcribe to `metaAccount` separately and then combine it with `walletConnectSessionRepository.sessionAccountsByMetaIdFlow` to always have the latest `metaAccount` state

File: feature-wallet-connect-impl/src/main/java/io/novafoundation/nova/feature_wallet_connect_impl/presentation/sessions/list/WalletConnectSessionsViewModel.kt [hunk 6]
```diff
@@ -33,6 +35,14 @@ class WalletConnectSessionsViewModel(
         router.openSessionDetails(WalletConnectSessionDetailsPayload(item.sessionTopic))
     }
 
+    private fun sessionsFlow(): Flow<List<WalletConnectSession>> {
+        return if (walletConnectSessionsPayload.metaId == null) {
+            interactor.activeSessionsFlow()
+        } else {
+            interactor.activeSessionsFlow(walletConnectSessionsPayload.metaId)
```
Comment (valentunn) [inline feature-wallet-connect-impl/src/main/java/io/novafoundation/nova/feature_wallet_connect_impl/presentation/sessions/list/WalletConnectSessionsViewModel.kt:42]: Maybe we can just create a single method in interface that will accept optional `metaId` parameter?

File: common/src/main/java/io/novafoundation/nova/common/base/BaseActivity.kt (hunk 2 above)
Comment (antonijzelinskij) [reply common/src/main/java/io/novafoundation/nova/common/base/BaseActivity.kt:47]: No, we subscribe in onCreate

Comment (antonijzelinskij) [reply common/src/main/java/io/novafoundation/nova/common/base/BaseActivity.kt:47]: It seems that the problem is in how github displays changes

File: feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/list/BalanceListViewModel.kt (hunk 4 above)
Comment (antonijzelinskij) [reply feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/list/BalanceListViewModel.kt:212]: It's number so I can't use isNotEmpty but I changed walletConnectAccountSessions to walletConnectAccountSessionCount to make clearer the responsibility

**Review by valentunn — APPROVED** [review]

**Pull request merged**